assign2/src/view/details/PostDetails/SmartPostDetailsView.tsx

```diff
@@ -0,0 +1,47 @@
+import {AppState} from "../../../model/Model";
+import {getQuestionAndAnswersWithVotes} from "../../../model/question/selectors";
+import Question from "../../../model/objects/Question";
+import React from "react";
+import {PostDetailsView} from "./PostDetailsView";
+import {connect} from "react-redux";
+import {Dispatch} from "redux";
+import {postDetailsPresenter} from "../../../presesnter/PostDetailsPresenter";
+import {VoteDirection} from "../../../model/objects/Vote";
+import User from "../../../model/objects/User";
+
+interface Props {
+    currentQuestion?: Question;
+    currentQuestionId: string;
+    onVote: (postId: number, postAuthorId: number) => (direction: VoteDirection) => void;
+}
+
+class SmartPostDetailsView extends React.Component<Props> {
+    render() {
+        return (
+            this.props.currentQuestion &&
+            <div>
+                <PostDetailsView question={this.props.currentQuestion} onVote={this.props.onVote}/>
+            </div>
+        );
+    }
+}
+
+
+function mapStateToProps(state: AppState, props: { currentQuestionId: string }) {
+    //FIXME Where should id be validated? We receive a string but it must be a number
+    const id = props.currentQuestionId;
```

Comment:
Meh, as it is passed via a prop, the easiest would be to just parse it here. But as you probably thought, you might hit situations in which the ID is not a number. Calling the presenter here to do a dispatch is not an option: doing dispatches in the mapStateToProps is a big no.

The most elegant solution would be to throw an error and catch it in an error boundary: https://reactjs.org/docs/error-boundaries.html